lib/components/form/submission-row.vue

```diff
@@ -0,0 +1,155 @@
+<!--
+Copyright 2017 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/opendatakit/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <!-- The frozen columns of the table -->
+  <tr v-if="questionColumns == null">
+    <td>{{ submission.__system.submitterName }}</td>
+    <td>{{ submissionDate }}</td>
+  </tr>
+  <!-- The rest of the table -->
+  <tr v-else>
+    <td v-for="column of questionColumns" :key="column.header"
+      :class="column.htmlClass"
+      :title="hasTitle(column) ? questionValue(column) : null">
+      <template v-if="column.type === 'binary'">
+        <span v-if="questionValue(column) !== ''"
+          class="icon-check-circle text-success">
+        </span>
+      </template>
+      <template v-else>
+        {{ questionValue(column) }}
+      </template>
+    </td>
+    <td>{{ submission.__id.replace(/^uuid:/, '') }}</td>
+  </tr>
+</template>
+
+<script>
+import { DateTime, Settings } from 'luxon';
+
+import { formatDate } from '../../util';
+
+const TITLE_QUESTION_TYPES = [
+  'string',
+  'select',
+  'select1',
+  'geotrace',
+  'geoshape',
+  'barcode'
+];
+
+export default {
+  name: 'FormSubmissionRow',
+  props: {
+    submission: {
+      type: Object,
+      required: true
+    },
+    // The question columns to display
+    questionColumns: Array // eslint-disable-line vue/require-default-prop
+  },
+  computed: {
+    submissionDate() {
+      return formatDate(this.submission.__system.submissionDate);
+    }
+  },
+  methods: {
+    // Returns the submission's value for the specified question column,
+    // formatting it according to the question's type.
+    questionValue(column) {
+      const rawValue = column.path.reduce((acc, name) => acc[name], this.submission);
+      if (rawValue == null) return '';
+      switch (column.type) {
```

Comment:
Note that `type` can be `undefined`.

Reply:
or `"unknown"` https://github.com/opendatakit/central-backend/blob/master/lib/data/schema.js#L54

Reply:
but it's more or less impossible that you'll actually get a data value for a field whose type is unknown.

Reply:
If we do encounter an `unknown` field, the current code will display nothing for a value that is `null` or `undefined`. Otherwise, it will display the value as a string without change.